compiler/cgen.nim

```diff
@@ -317,8 +317,13 @@ proc resetLoc(p: BProc, loc: var TLoc) =
       genObjectInit(p, cpsStmts, loc.t, loc, true)
     else:
       useStringh(p.module)
+
+      # array passed as argument decayed into pointer, bug #7332
+      let locName = if skipTypes(loc.t, abstractInst).kind == tyArray and loc.k == locParam:
```

Comment:
Can't it use ``getTypeDesc`` here always?

Reply:
it can, and no harm, only a bit ugly for generated code.
if you concern about compiler performance, then I will use only getTypeDesc